### ISSUE TITLE
Missing drakeConfig.cmake - Documentation Needed for Applications That Need to Link Against Drake's Libraries

### ISSUE THREAD

**Issue by liangfok** [RobotLocomotion/drake#3287]
# Problem Definition

I'd like to create a software project that is built using CMake and links against Drake's exported libraries. In my top-level CMakeLists.txt, I tried to use `find_package(drake)` but encountered the following error when running `cmake`:

```
CMake Warning at CMakeLists.txt:9 (find_package):
  By not providing "Finddrake.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "drake", but
  CMake did not find one.

  Could not find a package configuration file provided by "drake" with any of
  the following names:

    drakeConfig.cmake
    drake-config.cmake

  Add the installation prefix of "drake" to CMAKE_PREFIX_PATH or set
  "drake_DIR" to a directory containing one of the above files.  If "drake"
  provides a separate development package or SDK, be sure it has been
  installed.
```

I searched in `drake-distro` for files called `*Config.cmake` and `*-config.cmake` and confirmed that none existed for Drake, though numerous existed for other libraries like lcm, yaml, etc.
# Suggested Course of Action
## Step 3: Provide a "Hello World" Implementation of a Project that Uses Drake
# Other Comments

When [building Drake as a ROS package within a Catkin workspace](http://drake.mit.edu/from_source_ros.html), other ROS packages in the same workspace can link against Drake's libraries ~~by simply declaring `<build_depend>drake</build_depend>` within their `package.xml` files~~. I'm not sure how Catkin managed to get this to work.

**Update (2016/09/01):** I was mistaken. One cannot declare `<build_depend>drake</build_depend>` in `package.xml` when there are `.cc` files that need to be compiled. Otherwise, the same error will appear. However, the libraries exported by Drake _can_ be linked. See, for example, [this CMakeLists.txt](https://github.com/RobotLocomotion/drake/blob/3f1cebd68948737e15fa3f324f609553cae018b6/ros/drake_cars_examples/CMakeLists.txt#L38). I can only assume that Catkin enabled this ability to link against the libraries despite not declaring a build depend.

**Comment by jamiesnape** [comment]
FYI @mwoehlke-kitware.

**Comment by mwoehlke-kitware** [comment]
Yes, we definitely want to fix this, and it's on my to-do list for cleaning out all the pods junk. Unfortunately, adding proper exported packages to something the size of drake and with bunches of existing tweaks to how it creates libraries is not trivial. (But the ease of use for consumers is definitely worthwhile!)

FWIW, see e.g. LCM (the modern, CMake-ified version that we're using) for a good example of doing this right...

**Comment by liangfok** [comment]
I just realized that this may be a duplicate of #2693.

**Comment by david-german-tri** [comment]
From f2f: This is nearly ready.  Once it is, we should try to consider convincing OpenHumanoids to switch to it, so that we could drop pkg-config.

**Comment by david-german-tri** [comment]
(However, Director also needs to build against older SHAs of Drake.)

**Comment by david-german-tri** [comment]
`drake-config.cmake` is on master, though there are trouble reports like #5456.  I'm going to close this issue as partially fixed, and partially obsolete due to #3129.
